plugins/woocommerce/src/Blocks/BlockTypes/PaymentMethodIcons.php

```diff
@@ -0,0 +1,168 @@
+<?php
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Blocks\BlockTypes;
+
+use WP_Block;
+
+/**
+ * PaymentMethods class.
+ */
+class PaymentMethodIcons extends AbstractBlock {
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected $block_name = 'payment-method-icons';
+
+	/**
+	 * Get the frontend script handle for this block type.
+	 *
+	 * @param string $key Data to get, or default to everything.
+	 * @return array|string
+	 */
+	protected function get_block_type_script( $key = null ) {
+		return null;
+	}
+
+	/**
+	 * Get the frontend style handle for this block type.
+	 *
+	 * @return string[]
+	 */
+	protected function get_block_type_style() {
+		return array_merge( parent::get_block_type_style(), [ 'wc-blocks-packages-style' ] );
+	}
+
+	/**
+	 * Extra data passed through from server to client for block.
+	 *
+	 * @param array $attributes  Any attributes that currently are available from the block.
+	 */
+	protected function enqueue_data( array $attributes = [] ) {
+		parent::enqueue_data( $attributes );
+		$this->asset_data_registry->add( 'paymentMethodIcons', $this->get_enabled_card_types() );
+		$this->asset_data_registry->add( 'wooPaymentsEnabled', $this->is_woo_payments_enabled() );
+	}
+
+	/**
+	 * Render the block.
+	 *
+	 * @param array    $attributes Block attributes.
+	 * @param string   $content    Block content.
+	 * @param WP_Block $block      Block instance.
+	 * @return string Rendered block type output.
+	 */
+	protected function render( $attributes, $content, $block ) {
+		$output = '';
+
+		if ( $this->is_woo_payments_enabled() ) {
+			$output  = '<div class="wp-block-woocommerce-payment-method-icons">';
+			$output .= $this->render_card_types( $attributes );
+			$output .= '</div>';
+		}
+
+		return $output;
+	}
+
+	/**
+	 * Render the card types.
+	 *
+	 * @param array $attributes Block attributes.
+	 * @return string Rendered block type output.
+	 */
+	private function render_card_types( $attributes ) {
+		$output             = '';
+		$enabled_card_types = $this->get_enabled_card_types();
+		$number_of_icons    = max( 1, min( intval( $attributes['numberOfIcons'] ?? 5 ), count( $enabled_card_types ) ) );
+
+		foreach ( $enabled_card_types as $card_type => $card_data ) {
+			if ( $number_of_icons > 0 ) {
+				--$number_of_icons;
+			} else {
+				break;
+			}
+
+			$output .= '<div class="wp-block-woocommerce-payment-method-icons__item">';
+			$output .= '<span class="wp-block-woocommerce-payment-method-icons__icon" style="background-image: url(\'' . esc_url( $card_data['icon'] ) . '\');" role="img" aria-label="' . esc_attr( $card_data['name'] ) . '"></span>';
+
+			$output .= '</div>';
+		}
+
+		return $output;
+	}
```

Comment:
It may be me but I find this function a bit hard to understand. I'd suggest the following but let me know if that feels better for you:

```suggestion
	private function render_card_types( $attributes ) {
		$output             = '';
		$enabled_card_types = $this->get_enabled_card_types();
		$number_of_icons    =  min( intval( $attributes['numberOfIcons'] ?? 5 ), count( $enabled_card_types ) );
		
		for ( $i = 0; $i <= $number_of_icons; $i++ ) {
		    $output .= '<div class="wp-block-woocommerce-payment-method-icons__item">';
			$output .= '<span class="wp-block-woocommerce-payment-method-icons__icon" style="background-image: url(\'' . esc_url( $card_data['icon'] ) . '\');" role="img" aria-label="' . esc_attr( $card_data['name'] ) . '"></span>';
             $output .= '</div>';
		}

		return $output;
	}
```

Basically it switches from `foreach` to `for` loop which removes the need for `max( 1, ...` and also we don't need to track the iterator manually which `if else break`. Let me know what you think 🙌

Reply:
That makes sense, thanks! Done in https://github.com/woocommerce/woocommerce/pull/57181/commits/2993debd51e3d2045e5bb955bf2fcae8ac2a0d35.